torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -354,8 +489,52 @@ void RegisterIndexPutInBlocks(
       }
     }
   }
+
   orig_data->replaceAllUsesAfterNodeWith(
-      next_node->output(0)->node(), next_node->output(0));
+      next_node->output(0)->node(),
+      next_node->outputs().at(next_node->outputs().size() - 1));
+}
+
+// Register inplace op node inputs/outputs through the blocks.
+void RegisterInplaceNodeInBlocks(
+    Value* orig_data,
+    Value* new_inplace_node,
+    Node* block_node,
+    Block* outer_block,
+    Node* next_node) {
+  auto cur_node = next_node;
+
+  while (nullptr != cur_node) {
+    if (cur_node->kind() != prim::Loop && cur_node->kind() != prim::If)
+      return;
+    cur_node = cur_node->owningBlock()->owningNode();
+  }
+
+  for (auto block_input : outer_block->inputs()) {
+    if (block_input->debugName() == orig_data->debugName()) {
```

Comment:
are there plans to support more than one inplace mutation on same object?

Reply:
Yes, this needs to be added. WIll leave a comment.